### DISCUSSION TITLE
Touch Gestures?

### DISCUSSION THREAD

**Discussion by quadruple-output** [emilk/egui#277] [Q&A] [answered]
Thanks to the amazing egui_template, I now have my first wasm project online – thank you for this great library!

My app has a drawing canvas which can be zoomed in and out by holding down CTRL/CMD when moving the mouse. While this works well on a PC after you get used to it, I quickly learned that it is not a concept which can ever work on a mobile device ;) .

Investigating the egui source, I could not find any hint that touch gestures are supported. Are there any plans for it? What would be needed to make it work? With some guidance, I might consider contributing.

**Comment by emilk** [comment, marked as answer]
Great question! `egui` does not currently support touch gestures, but it should be fairly easy to add. I created an issue with an outline of a plan: https://github.com/emilk/egui/issues/279

**Reply by quadruple-output** [reply]
Thanks! I had a look and I think I might give it a try. Would be my first pull request ever, so I guess we will need a few round trips. I'll try my best, though…
